hooks/environment

```diff
@@ -200,6 +200,14 @@ function login_using_aws_ecr_get_login_password() {
   # amend the ~~~ log heading with ^^^ to add the AWS account IDs
   echo "^^^ Authenticating with AWS ECR in $region for ${account_ids[*]} :ecr: :docker:"
 
+  local ecr_registry_url;
+  if [[ ${region:0:3} == "cn-" ]]; then
+    # the ecr registry in China regions has a different URL
+    ecr_registry_url="$account_id.dkr.ecr.$region.amazonaws.com.cn"
+  else
+    ecr_registry_url="$account_id.dkr.ecr.$region.amazonaws.com"
+  fi
```

Comment:
Hey! Tests are failing because you are using `account_id` before it was declared. If you move this block inside the `for` (or at least in the `else` where you are using it) it should work.